R/tar-terra-rast.R

```diff
@@ -145,6 +149,9 @@ tar_rast_read <- function(preserve_metadata) {
             zip::unzip(zipfile = path, exdir = tmp)
             terra::rast(file.path(tmp, basename(path)))
         },
+        gdalraster_sozip = function(path) {
+          terra::rast(paste0("/vsizip/{", path, "}/"), basename(path))
+        },
```

Comment:
Do we need to worry about file.path here, or is this syntax going to be generalisable to all OS's?